server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -165,6 +237,18 @@ public static FetchSourceContext fromXContent(XContentParser parser) throws IOEx
                         includes = new String[] { parser.text() };
                     } else if (EXCLUDES_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
                         excludes = new String[] { parser.text() };
+                    } else if (INCLUDE_VECTORS.match(currentFieldName, parser.getDeprecationHandler())) {
+                        includeVectors = parser.booleanValue();
+                    } else {
+                        throw new ParsingException(
+                            parser.getTokenLocation(),
+                            "Unknown key for a " + token + " in [" + currentFieldName + "].",
+                            parser.getTokenLocation()
+                        );
+                    }
+                } else if (token == XContentParser.Token.VALUE_BOOLEAN) {
+                    if (INCLUDE_VECTORS.match(currentFieldName, parser.getDeprecationHandler())) {
+                        includeVectors = parser.booleanValue();
```

Comment:
We also parse source in MultiGet operation defined in here https://github.com/elastic/elasticsearch/blob/19257125b19ff05a6b014f37693f9417ca579626/server/src/main/java/org/elasticsearch/action/get/MultiGetRequest.java#L473-L490

Should we replace the duplicate existing parsing logic with our current approach defined in `FetchSourceContext`?